### ISSUE TITLE
poisoned with slowed not shown on statusbar

### ISSUE THREAD

**Issue by gfgtdf** [wesnoth/wesnoth#3197]
when  unit is slowed and poisoned only the slowed icon is shown in the right side panel. a test showed that i was casued by https://github.com/wesnoth/wesnoth/commit/d51ccbe5386e5325c6ecfe6c46472f7a8113bfa7#diff-33ec5f3f8d3f054d6defb5e8fb113cf0

**Comment by CelticMinstrel** [comment]
So maybe change `[unit_status]` to use `ref=unit-box-topright` instead of `ref=label-hp`? And obviously update the rect accordingly.

**Comment by gfgtdf** [comment]
> And obviously update the rect accordingly.

ye i have no idea how that works, he wiki page https://wiki.wesnoth.org/ThemeWML just says that rect  takes a value of the form a,b,c,d with no hint what a b c and d actually do.

**Comment by CelticMinstrel** [comment]
Um, what? It explains it literally right there on the next lines.

**Comment by gfgtdf** [comment]
no it just doesn't, its complteley unclear which of those values arer are x values which are y valuess, from which corner the positions are measured etc.

**Comment by CelticMinstrel** [comment]
Ah, I see what you mean. It never actually says whether it's using th`x1,y1,x2,y2` format or the `top,left,bottom,right` format.

I think it's the former? Should be possible to figure out though by looking at the existing themes.

**Comment by Landiss** [comment]
I agree wiki is not very clear about this. As I have recently made my own theme with larger font and increased side panel width, I can perhaps explain it.

Basically rec="a,b,c,d" creates a rectangle, where a,b are top left x,y coordinates and c,d are bottom right coordinates. So, for example rec="0,10,20,30" would create a rectangle that has top left corner at the very left side of the screen, 10 pixels below the very top and is 20 pixels wide and tall.

However, there are 3 ways the values can be inputted:
- if the values are not preceded with anything, they are absolute values. Meaning they don't take anything from ref.
- if the value is preceded with =, than it is taken from the absolute value in ref (a is ref a, b is ref b etc.), it can also be =+ or =- that changes that value by the specified amount
- if the value is preceded only with + (or -), it's a special case. a becomes ref c + the value, b becomes ref d + the value, while c and d become rec (not ref) a and b + the values, in other words, they become width and height. Note that +0 is also possible and used often for a and/or b, as it takes lower-bottom corner of the ref rectangle.


And about this specific issue - changing ref to unit box won't fix anything and would risk having the status icon overlap with the HP phrase in some translations.

Do we know how much space is required to display 2 icons? They are displayed one after another?

Currently the HP label rectangle is 60 pixels wide and the status icon is displayed another 10 pixels to the right of the end of it. 

If I count right, that is currently 7 (left margin) + 72 (width of the unit box) +3 (unit box right border, if I'm reading that right) + 4 (space between unit box and HP label) + 60 (HP label width) + 10 (another margin) = 146. This is counted from the left side of the side panel. The panel width is 182, but the unit_status ends 2 pixels from the right side, so it currently has width of 182-2-146 = 34.

Changing the hp width to 38 and the space between it and unit_space to 3 shows both icons, but to be honest, it's not beautiful, especially for languages that have shortcut for HP (which seems to be... all but Czech?) and especially when there is just one icon. See the screenshots:
![2 icons](https://user-images.githubusercontent.com/20052491/41130236-07561a78-6ab6-11e8-88d0-1d672d886c17.PNG)

![en 1 icon](https://user-images.githubusercontent.com/20052491/41130322-6f6420c4-6ab6-11e8-8f95-c0a4c35950d6.PNG)


TL;DR: it's possible to fit both icons by changing values slightly, however bad it might look. Proper fix would be to redesign the theme and increase width of the side panel. Anyway, the quick fix is the following.

`id=label-hp` should have ` rect="+4,=-1,+38,+14"` instead of `rect="+4,=-1,+60,+14"` and `unit_status` should have `rect="+3,=-4,1022,+16"` instead of `rect="+10,=-4,1022,+16"`

**Comment by CelticMinstrel** [comment]
What about showing the icons in a column rather than a row?

**Comment by Landiss** [comment]
That's certainly a good idea, but I don't think it be done via themeWML itself, because the individual icons are not exposed to it.

Also, worth keeping in mind - looks like there are actually 4 statuses theoretically possible in mainline, although it would be very unusual for more than 2 to be displayed at the same time. They are invisible, petrified, slowed and poisoned. However, in UMC I think it might be possible to get more than 2 at the same time. So, it would make sense to have a solution for more than 2 icons.

Also worth noting that there is also the classic theme, for which showing the icons in column would not make sense.

**Comment by CelticMinstrel** [comment]
User-defined statues are why I suggested a column, but yeah I guess that might require some change in the C++?

**Comment by Landiss** [comment]
I don't really know for sure, but that's what I think, yes.

Also, for reference, the icons are 24x24. The default theme should be able to fit 3 icons in a column. The column would not work for the classic theme though, it has to be a row and max 2 icons.

A different solution would be to get rid of icons and change them to text, similar to how traits are done.

**Comment by CelticMinstrel** [comment]
I don't think switching to text is a good idea; in fact we should really be using icons for the attacks.

**Comment by CelticMinstrel** [comment]
Yes indeed! If you use icons you shouldn't need the words, ie the icons can replace the "ranged-pierce" line (maybe include it in a tooltip though). As for the actual attack icon, I think that should be quite a bit larger than in your screenshot, though there's kinda not much space in the theme... might be better to leave it out for now? Ideally I think you'd have the icon taking up almost full size, with all the information about that attack to the right of the icon (including its name and icons for range and type). But I'm not sure if the ThemeWML can do that within a single report? (And if weapon icons were a separate report you'd have alignment issues, because a weapon description's length depends on the number of specials.)

**Comment by jostephd** [comment]
Yes, the black and red icons have tooltips.

I'm not sure about removing the words "ranged-pierce" from the text part of the report. It could work for mainline, but a UMC era that uses custom ranges or damage types would need to include icons for them. Would that be an obstacle for UMC authors? We should be making things easy for them.

About the attack icons size, at first I tried to make it larger but I couldn't figure out how to do the theme layout, for the reasons you guessed: there is little room for it and I couldn't figure out a way to put the attack icon to one side and have the weapon specials to its left or right. In the end I decided to just make it smaller, since the important information is the black icon, the red icon, and the two numbers after them (I remember the damage per hit and number of hits of most units, but I couldn't tell you what the attacks are called).

I think I prefer the report with the attack icons than without, they're unobtrusive.

I'll open an orderly PR for this later. Maybe also add the icons to unit_preview_pane and the generated help pages.

**Comment by jostephd** [comment]
About the attack icons and layout, it's easy to make it look like this:

![screenshot_2018-11-18_05-09-14](https://user-images.githubusercontent.com/24784687/48668937-55f57380-eaf0-11e8-8fd3-e31376829046.png)

Does anyone know how to make it look like this:

![screenshot_2018-11-18_05-09-14](https://user-images.githubusercontent.com/24784687/48668962-ff3c6980-eaf0-11e8-92d7-3d8290f2a3f9.png)
